From 7b335df2b2375442630d0ae12323b61b99a88d64 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Fri, 28 Sep 2018 10:40:09 -0500
Subject: [PATCH 4749/5242] MLK-18997: mtd: fsl-quadspi: disable the TDH DDR
 enable bit if it's been set

commit  06b8684638efab1c016d22a7bea822c5d77ad132 from
https://source.codeaurora.org/external/imx/linux-imx.git

for 6UL/7D, the TDH bit should only be set when DDR mode enabled. This
bit cannot be cleared during module reset, so check this bit in
nor_setup to make sure it cleard and won't affect the following
operations.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index e7a9b9c..782837d 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -840,6 +840,10 @@ static int fsl_qspi_nor_setup(struct fsl_qspi *q)
 			| QUADSPI_SMPR_HSENA_MASK
 			| QUADSPI_SMPR_DDRSMP_MASK), base + QUADSPI_SMPR);
 
+	/* disable the TDH bit of FLSHCR if it's been set*/
+	reg = qspi_readl(q, base + QUADSPI_FLSHCR);
+	qspi_writel(q, reg & ~QUADSPI_FLSHCR_TDH_MASK, base + QUADSPI_FLSHCR);
+
 	/* Enable the module */
 	qspi_writel(q, QUADSPI_MCR_RESERVED_MASK | QUADSPI_MCR_END_CFG_MASK,
 			base + QUADSPI_MCR);
-- 
1.7.9.5

